go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -772,7 +757,12 @@ func (h *defaultNetworkControllerEventHandler) AddResource(obj interface{}, from
 
 	case factory.EgressNodeType:
 		node := obj.(*kapi.Node)
-		return h.oc.reconcileNodeForEgressIP(nil, node)
+		// Update node in zone cache; value will be true if node is local
+		// to this zone and false if its not
+		h.oc.eIPC.nodeZoneState.LockKey(node.Name)
+		h.oc.eIPC.nodeZoneState.Store(node.Name, h.oc.isLocalZoneNode(node))
```

Comment:
lock here is to protect the write we are doing into `nodeZoneState`

Reply:
ack. Then my source of confusion is that SyncMaps have a lock and that should be enough.
But I can now see that this is not the case. Here. All good.

ref: https://github.com/ovn-org/ovn-kubernetes/pull/3097

```
// LockKey should be called before reading/writing entry value,
// it guarantees exclusive access to the key.
// Unlock(key) should be called once the work for this key is done to unlock other threads
// After the key is unlocked there are no guarantees for the entry for given key
func (c *SyncMap[T]) LockKey(key string) {
...

// Store sets the value for a key.
// If key-value was already present, it will be over-written
func (c *SyncMap[T]) Store(lockedKey string, newEntry T) {
...
```